knowledge-base/chart-datatable-binding.md

```diff
@@ -0,0 +1,78 @@
+---
+title: How to use DataTable as source of data for the Chart
+description: How to convert DataTable to List and use it as data source for the Chart
+type: how-to
+page_title: How to use DataTable as data source for the Chart
+slug: chart-datatable-binding
+position: 
+tags: 
+ticketid: 1589460 
+res_type: kb
+---
+
+## Environment
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product</td>
+	    	<td>Charts for Blazor</td>
+	    </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+I want to use DataTable as a data source for the Chart series.
+
+
+## Solution
+
+The Telerik UI for Blazor Chart does not support DataTable binding out of the box. To use DataTable instance as a data source for the series of the Chart you have to convert the DataTable to IEnumerable<T> e.g. List<T>.
+
+Here is a simple example:
+
+````CSHTML
+@using System.Data;
+@using System.Dynamic;
+
+<TelerikChart>
+    <ChartSeriesItems>
+        <ChartSeries Type="ChartSeriesType.Pie"
+                     Data="Data"
+                     Field="Value"
+                     CategoryField="Name">
+            <ChartSeriesLabels Visible="true"></ChartSeriesLabels>
+        </ChartSeries>
+    </ChartSeriesItems>
+</TelerikChart>
+
+@code {
+    private List<ExpandoObject> Data { get; set; } = ConvertDataTable(GetData(), "Value", "Name");
+
+    private static DataTable GetData()
+    {
+        var table = new DataTable();
+        table.Columns.Add("Value", typeof(double));
+        table.Columns.Add("Name", typeof(string));
+        table.Rows.Add(1, "John");
+        table.Rows.Add(5, "Peter");
+        table.Rows.Add(6, "Paul");
+        return table;
+    }
+
+    private static List<ExpandoObject> ConvertDataTable(DataTable dt, string field, string categoryField)
+    {
+        List<ExpandoObject> data = new List<ExpandoObject>();
+        foreach (DataRow row in dt.Rows)
+        {
+            var expando = new ExpandoObject() as IDictionary<string, object>;
+            expando.Add(field, row[field]); 
+            expando.Add(categoryField, row[categoryField]);
+            data.Add((ExpandoObject)expando);
+        }
+        return data;
+    }
+}
+````
+
```

Comment:
Consider a "See Also" section with other useful articles or demos related to Chart binding. Optionally, consider linking to this article from a Chart docs article, but only if you find a suitable article and place.